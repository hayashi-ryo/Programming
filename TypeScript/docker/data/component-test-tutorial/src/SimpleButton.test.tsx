import { render, screen } from "@testing-library/react";
import { SimpleButton } from "./SimpleButton";
import userEvent from "@testing-library/user-event";

test("Check Button", async () => {
  // コンポーネントテスト
  render(<SimpleButton />); // 描画
  const simpleButton = screen.getByRole("button"); // 初期値の確認
  expect(simpleButton).toHaveTextContent("OFF");
  await userEvent.click(simpleButton); // ONに変化することの確認
  expect(simpleButton).toHaveTextContent("ON");
});

test("Check Button", async () => {
  // スナップショットテスト
  const view = render(<SimpleButton />);
  expect(view.container).toMatchSnapshot();
});
