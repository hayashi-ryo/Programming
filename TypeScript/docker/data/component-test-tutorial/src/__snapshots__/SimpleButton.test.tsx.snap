// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check Button 1`] = `
<div>
  <button>
    OFF
  </button>
</div>
`;
